### PR TITLE
add unified rollback service

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**

### CODE REVIEW

**Review by Mingela — APPROVED** [review]

File: btc-withdrawal/src/main/kotlin/com/d3/btc/withdrawal/service/BtcRollbackService.kt [hunk 1]
```diff
@@ -26,29 +24,30 @@ class BtcRollbackService(
...
-            )
-        }.fold(
+        val withdrawalFinalizationDetails = WithdrawalFinalizationDetails(
+            satToBtc(withdrawalDetails.amountSat),
+            BTC_ASSET,
+            satToBtc(withdrawalDetails.withdrawalFeeSat),
+            BTC_ASSET,
+            withdrawalDetails.sourceAccountId,
+            withdrawalDetails.withdrawalTime,
+            withdrawalConsumer.creator
```
Comment (dolgopolovwork) [inline btc-withdrawal/src/main/kotlin/com/d3/btc/withdrawal/service/BtcRollbackService.kt]: This value stands for `destinationAddress`. Not creator.

File: btc-withdrawal/src/main/kotlin/com/d3/btc/withdrawal/service/BtcRollbackService.kt [hunk 2]
```diff
@@ -26,29 +24,30 @@ class BtcRollbackService(
...
+        )
+
+        rollbackService.rollback(withdrawalFinalizationDetails, reason).fold(
             {
                 logger.info {
-                    "Rollback(accountId:${withdrawalDetails.sourceAccountId}, amount:${satToBtc(amountToRollbackSat).toPlainString()}) was committed"
+                    "Rollback(accountId:${withdrawalDetails.sourceAccountId}, amount:${satToBtc(
```
Comment (dolgopolovwork) [inline btc-withdrawal/src/main/kotlin/com/d3/btc/withdrawal/service/BtcRollbackService.kt]: Could you please add information about fees as well?

File: btc-withdrawal/src/main/kotlin/com/d3/btc/withdrawal/service/BtcRollbackService.kt (hunk 1 above)
Comment (Alexey-N-Chernyshov) [reply btc-withdrawal/src/main/kotlin/com/d3/btc/withdrawal/service/BtcRollbackService.kt]: toAddress is a BTC address here, so only one option to get withdrawal service client id is to get it from consumer

File: btc-withdrawal/src/main/kotlin/com/d3/btc/withdrawal/service/BtcRollbackService.kt (hunk 2 above)
Comment (Alexey-N-Chernyshov) [reply btc-withdrawal/src/main/kotlin/com/d3/btc/withdrawal/service/BtcRollbackService.kt]: done

**Review by dolgopolovwork — APPROVED** [review]

**Pull request merged**